From 1f612f27cefba3cce02f1c00df17c70096c243a1 Mon Sep 17 00:00:00 2001
From: Daniel Cordero <theappleman@gmail.com>
Date: Sat, 6 Dec 2008 10:41:42 +0000
Subject: [PATCH] t9129: Make test find any UTF-8 locale

Commit 16fc08e2d86dad152194829d21bc55b2ef0c8fb1 introduced a
test that failed if the hardcoded en_US.UTF-8 locale was not
installed.

If no UTF-8 locale is found, skip the tests.
---
 t/t9129-git-svn-i18n-commitencoding.sh |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/t/t9129-git-svn-i18n-commitencoding.sh b/t/t9129-git-svn-i18n-commitencoding.sh
index 8a9dde4..6ac50ba 100755
--- a/t/t9129-git-svn-i18n-commitencoding.sh
+++ b/t/t9129-git-svn-i18n-commitencoding.sh
@@ -6,6 +6,11 @@ test_description='git svn honors i18n.commitEncoding in config'

 . ./lib-git-svn.sh

+utf8_locale=$(locale -a | sed -ne '/[Uu][Tt][Ff]-*8/{
+	p
+	q
+'})
+
 compare_git_head_with () {
	nr=`wc -l < "$1"`
	a=7
@@ -15,7 +20,8 @@ compare_git_head_with () {
 }

 compare_svn_head_with () {
-	LC_ALL=en_US.UTF-8 svn log --limit 1 `git svn info --url` | \
+	say "Using locale: $utf8_locale"
+	LC_ALL="$utf8_locale" svn log --limit 1 `git svn info --url` | \
		sed -e 1,3d -e "/^-\{1,\}\$/d" >current &&
	test_cmp current "$1"
 }
@@ -60,7 +66,7 @@ do
	'
 done

-if locale -a |grep -q en_US.utf8; then
+if test "$utf8_locale"; then
	test_expect_success 'ISO-8859-1 should match UTF-8 in svn' '
	(
		cd ISO-8859-1 &&
@@ -78,7 +84,7 @@ if locale -a |grep -q en_US.utf8; then
		'
	done
 else
-	say "UTF-8 locale not available, test skipped"
+	say "UTF-8 locale not available, tests skipped"
 fi

 test_done
--
1.6.1.151.g15260
